jobs/buildhub/to_kinto.py

```diff
@@ -155,16 +102,53 @@ def fetch_existing(
             )
             previous_run_etag = '"%s"' % highest_timestamp
 
-    new_records = client.get_records(
-        _since=previous_run_etag,
-        pages=float('inf')
-    )
-
-    for record in new_records:
-        records[record['id']] = [
-            record['last_modified'],
-            hash_record(record)
-        ]
+    # The reason we can't use client.get_records() is because it is not
+    # an iterator and if the Kinto database has 1M objects we'll end up
+    # with a big fat Python list of 1M objects that has repeatedly caused
+    # OOM errors in our stage and prod admin nodes.
+    if previous_run_etag:
+        # However, we can use it if there was a previous_run_etag which
+        # means we only need to extract a limited amount of records. Not
+        # the whole Kinto database.
+        new_records_batches = [client.get_records(
+            _since=previous_run_etag,
+            pages=float('inf')
+        )]
+    else:
+        def new_records_iterator():
+            params = {'_since': None}
```

Comment:
params is thus useless, no?

Reply:
I don't understand your comment. It's set to `{'_since': None}` for the first request. Then, for all the other paginated requests it's `{}`.